packages/connect/src/device/Device.ts

```diff
@@ -1170,12 +1171,13 @@ export class Device extends TypedEmitter<DeviceEvents> {
         return null;
     }
 
-    async dispose() {
+    dispose() {
         this.removeAllListeners();
         if (this.session && this.lastAcquiredHere) {
             try {
-                await this.cancelableAction?.();
-                await this.commands?.cancel();
+                // note for review: these deleted lines might not be needed anymore - in case of reload we don't have enough time
+                // anyway, so seems to be enough to release session so that we don't get 'unacquired' device after reloading. If
+                // anything stays on the screen, it will get flushed away by the initial Cancel (see device run method)
```

Comment:
This seems true. The only thing `commands.cancel()` does is setting `disposed` flag and the only thing `cancelableAction()` may do is to send Cancel to device, which is now done in `runInner`.

In that case, this try/catch can be completely removed and `this.transport.release` returned directly.